### ISSUE TITLE
Add docs for missing Azure remote options

### ISSUE THREAD

**Issue by shcheklein** [iterative/dvc.org#5151]
For azure remote:

```python
               "exclude_environment_credential": Bool,
               "exclude_visual_studio_code_credential": Bool,
               "exclude_shared_token_cache_credential": Bool,
               "exclude_managed_identity_credential": Bool,
```

are missing from docs, but critical in some environments. Docs would have saved me a lot of time. At least this ticket hopefully can also help other folks who need to make DVC avoid certain credentials.

**Comment by aayushk9** [comment]
I suggest documenting the Azure configuration options in the User Guide section of our documentation. This section is dedicated to detailed explanations of tool features and mechanisms, making it an ideal place for users to understand and customize Azure Remote behavior.

**Comment by aayushk9** [comment]
I can solve this issue.

**Comment by shcheklein** [comment]
yes, that's the right place! Please feel free to take this @aayushk9 - I can help to iterate on it and give you more details.

Thank!
